### ISSUE TITLE
Looking for a new maintainer

### ISSUE THREAD

**Issue by jgonggrijp** [jgonggrijp/pip-review#76]
I'm about to release version 1.1.0 of pip-review. In addition to features and bugfixes that were requested or contributed in the past few years, that release will include a notice that I'm looking for a new maintainer to take over.

Until there is a new maintainer, at least in the foreseeable future, I will continue to accept pull requests and release new versions to PyPI once in a while. However, please be prepared that my time investment in pip-review will continue to be as slim as it has been for the past few years.

The reason my time investment in pip-review declined, is that I'm no longer using it myself. For the same reason, I now feel that I'm not the right person to keep this tool in the air.

pip-review is a very small and relatively simple project. If you are actively using pip-review and you don't mind adopting it, please submit a pull request and mention in the pull request that you're interested in taking over.

**Comment by Jacksole** [comment]
While, I'm not the prolific Python programmer, I'm willing to take a shot at taking over or assist in maintaining the program. @jgonggrijp

**Comment by jgonggrijp** [comment]
Thank you for the kind offer, @Jacksole! Let's ride together in this for a while and see what happens.

Other people are still welcome to express their interest as well.

**Comment by Jacksole** [comment]
Ok, thanks @jgonggrijp . So what would you need me to do?

**Comment by jgonggrijp** [comment]
Perhaps pick an issue from the list and see what you can do about it. Maybe you can figure out the issue that you reported yourself.

**Comment by jgonggrijp** [comment]
@Jacksole also, please take your time and feel free to ask questions. I'm not in a hurry and it is okay to be new at this.

**Comment by Jacksole** [comment]
@jgonggrijp Thanks

**Comment by berislavlopac** [comment]
@jgonggrijp Have you tried actually pitching the project to PyPA? It looks like something that might make sense as actual part of `pip` itself...

**Comment by jgonggrijp** [comment]
@berislavlopac Thanks for the suggestion. I think the odds are bleak, though. pip-review was *removed* from pip-tools. I just adopted it because I didn't want it to disappear. Even pip-tools, which is much more sophisticated and has a much larger user base than pip-review, is not a part of pip, even after many years of existence. Although it is a building block of pipenv.

**Comment by jonpalle** [comment]
@jgonggrijp you mentioned that you are not using pip-review anymore. May I ask why that is? I think it is a neat little tool 😃

**Comment by jgonggrijp** [comment]
@jonpalle Of course you may ask. Nowadays, when developing a Python project, I always create a dedicated virtualenv for that project. I pin the dependencies using pip-tools and I rarely update the packages in the virtualenv, except when an automated dependency tracker like @dependabot warns me about a security issue (in which case I'm updating only one dependency, by accepting a pull request) or when dusting off a stale project. In the latter case, `pip-compile -U ; pip-sync` does a much cleaner job than `pip-review -a`; the former produces a coherent set of compatible dependencies and removes packages that are no longer needed, while the latter will eventually accumulate obsolete packages and might install incompatible packages side by side.

Of course, I could do `pip-review -i` instead of `pip-review -a`, but it feels like more effort. I think it could be sensible to use `pip-review -i` for user-wide or system-wide packages, but I happen to use MacPorts for that already.

**Comment by jonpalle** [comment]
Thanks @jgonggrijp for enlightening me (and hopefully others)! 😉

**Comment by jgonggrijp** [comment]
Which reminds me: are you still interested in taking over, @Jacksole?

**Comment by Jacksole** [comment]
> Which reminds me: are you still interested in taking over, @Jacksole?

Yes, I am.

**Comment by jgonggrijp** [comment]
Great to hear that!

**Comment by Jacksole** [comment]
Ok so how does transfer of ownership work exactly?

**Comment by jgonggrijp** [comment]
In this case, I suggest you just start doing maintenance work, i.e. pick any issue that you find important/easy/fun enough, fix it and submit a PR. Rinse and repeat.

At some point, I'll be convinced that not all work you submit needs to be reviewed by me and I'll give you collaborator access. A bit further in the future, I'll also give you the rights to publish the package to PyPI. Finally (when it's clear that you don't need me at all), I'll transfer the repository to you entirely so you become the owner.

Until that time, please feel very welcome to bombard me with questions. It is in my own interest that you have all your questions answered.

As I said before, other people are welcome to enroll in this same "promotion track" at the same time. If we end up with _two_ new maintainers, that's totally fine by me.

**Comment by ssbarnea** [comment]
I doubt pip-review is popular enough for PyPA organization to take it under its wing, at least for now. Clearly for longer term maintenance it needs an organization. One possible option would be under https://github.com/pycontribs/ which is generic. Later if it becomes more popular and better maintained we could propose transferring it again.

**Comment by jgonggrijp** [comment]
@ssbarnea Thanks for the suggestion. Is the intent of PyContribs similar to [Jazzband](https://github.com/jazzband)? You may want to consider joining forces with them, since they have a much larger team.

I have thought about moving pip-review to Jazzband. So far, I opted out of it, firstly because pip-tools is already in Jazzband, which pip-review is a removed feature of, and secondly because I got the impression I would still have to take some responsibility for pip-review and/or other Jazzband projects. I'll ponder it some more, though.

@Jacksole how would you feel about pip-review joining Jazzband or a similar organization? You could probably still take over maintainership in that case, but it would require that you become a member of the organization in question.

**Comment by ssbarnea** [comment]
Yep same intent. I kinda see JazzBand focused on playing mostly only on djangos ;) -- while I was aware of the organization I did not know they made the joining so easy,  aka automated. I just joined too few minutes ago. Clearly they did spent far more time than me on pycontribs. One thing I should maybe consider in the future is merging the two. Meanwhile I need to discover who is behind, I am sure there are not >900 owners. 

I am sure the either one of them would be better than an individual account or even a new organization. In fact I would personally advise against creating a new organization for such a small project, mainly because because of the extra maintenance. I am also member on few other python specific organization with similar goals by those are aimed for specific project types like @tox-dev @PyCQA @pytest-dev @sphinx-contrib (their name should be self explanatory).

Any org that has 3-4 or more owners and shares same goal should be fine.

**Comment by ssbarnea** [comment]
On the other note, I just observed that this project is forked from pip-tools. The question is why this new cli was not added to pip-tools itself?

I recently become one pip-tools maintainer and I will support a change that contributes it to pip-tools. Once done we can archive this repository. Less packages and better maintained may address the current problem.

**Comment by flamableconcrete** [comment]
@ssbarnea: `pip-review` was part of `pip-tools` originally, but was [split out of the main project back around 2015](https://github.com/jazzband/pip-tools/issues/185). This is my short observation as an observer. I don't use either tool, and don't even remember why I was watching this issue in the first place, but hopefully that gives some context!

**Comment by ssbarnea** [comment]
@flamableconcrete Well, while I do not know the reasons for the split, I can see which one is better maintained ;)

**Comment by someparsa** [comment]
Hi, are you still looking for a maintainer? I am so much interested, but I am not a programming specialist. With gradual advices, I can be helpful, I believe. Please let me know.

**Comment by jgonggrijp** [comment]
Yes, I am still looking for a maintainer. Multiple is also fine. If you want to give it a shot, you are more than welcome; you can start by making any code contribution. *Of course* I will guide you through it!

**Comment by someparsa** [comment]
Thank you Julian, that's great. I forked your repo and added a feature to print upgradable packages in tabular format. I think it is pretty indeed.

Please have in mind, I am not a software expert, nor I have maintenance experience. I should do lots of searches to do the tasks. But at the end of my coding tries and errors, I code something without errors :)) It would be great if you could have look over. Just use this to run: `python -m pip_review`**`-t`** in the command, after you locate my updates. thank you

**Comment by jgonggrijp** [comment]
Great. Could you submit a pull request for the new feature, so I don't have to go searching for it? GitHub has documentation on how to do that, with links to explanations on other relevant topics, over [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).

**Comment by someparsa** [comment]
Sure Julian, will submit it asap. BTW, based on the issues raised here and all ideas come to mind, we can make a prioritized to-do list to have a road map for all interested contributors. Also we need a doc and wiki page for the package. Gradually will work on these as well. take care :)

**Comment by jgonggrijp** [comment]
@Jacksole The discussion in #99 made me realize I could have given you more guidance back in 2020. Would you like to have another shot at this?
